### PR TITLE
PR: Enhancement to date filters

### LINKED ISSUE #735: Enhancements to date filter
Right now the date filter only allows the user to choose between posts that are older than xx days or published within the last xx days.

Following are additional features that can be added to this filter

- [x] Ability to select a specific publish date/day using a date selector. For example all posts published on the 11th of January 2019
- [x] Ability to select a range - a start and end date using date selector, so you can pinpoint the publish date range for example 1 until 11th of January 2019.

### PR DESCRIPTION
**Pull request by esudalaimadan**
Fix #735

### CODE REVIEW

File: include/Core/Base/Mixin/Renderer.php [hunk 1]
```diff
@@ -629,7 +629,25 @@ protected function render_filtering_table_header() {
...
+			<td scope="row" colspan="3">
+				<input name="smbd_<?php echo $slug; ?>_restrict" id="smbd_<?php echo $slug; ?>_restrict" value="true" type="checkbox">
+				<label for="smbd_<?php echo $slug; ?>_restrict"><?php printf( __( 'Only restrict to %s which are ', 'bulk-delete' ), $item ); ?></label>
+				<select name="smbd_<?php echo $slug; ?>_op" id="smbd_<?php echo $slug; ?>_op" disabled>
+					<option value="before"><?php _e( 'older than', 'bulk-delete' ); ?></option>
+					<option value="after"><?php _e( 'posted within last', 'bulk-delete' ); ?></option>
+					<option value="="><?php _e( 'published on', 'bulk-delete' ); ?></option>
+					<option value="between"><?php _e( 'published between', 'bulk-delete' ); ?></option>
+				</select>
+				<div style="display:inline;" id="smbd_<?php echo $slug; ?>_days_box"><input type="number" name="smbd_<?php echo $slug; ?>_days" id="smbd_<?php echo $slug; ?>_days" class="screen-per-page" disabled value="0" min="0"><?php _e( 'days', 'bulk-delet' ); ?></div>
```
Comment (sudar) [inline include/Core/Base/Mixin/Renderer.php]: There is a typo in the second parameter of `_e_`

File: include/Core/Base/Mixin/Renderer.php [hunk 2]
```diff
@@ -629,7 +629,25 @@ protected function render_filtering_table_header() {
...
+				<input type="text" name="smbd_<?php echo $slug; ?>_pub_date" id="smbd_<?php echo $slug; ?>_pub_date" placeholder="Choose Date" style="display: none;">
+				<input type="text" name="smbd_<?php echo $slug; ?>_pub_date1" id="smbd_<?php echo $slug; ?>_pub_date1" placeholder="Choose Date" style="display: none;">
```
Comment (sudar) [inline include/Core/Base/Mixin/Renderer.php]: - Let's rename `pub_date1` to `pub_date_before`
- Let's rename `pub_date2` to `pub_date_after`

This change needs to be also done in the places where we are consuming these variables.

**Review by sudar — CHANGES_REQUESTED** [review]

File: include/Core/Base/Mixin/Renderer.php (hunk 1 above)
Comment (esudalaimadan) [reply include/Core/Base/Mixin/Renderer.php]: Fixed.

File: include/Core/Base/Mixin/Renderer.php (hunk 2 above)
Comment (esudalaimadan) [reply include/Core/Base/Mixin/Renderer.php]: Renamed the following:
`pub_date1` => `pub_date_start`
`pub_date2` => `pub_date_end`

**Review by sudar — APPROVED** [review]

**Pull request merged**